Marlin/stepper.cpp

```diff
@@ -328,13 +329,42 @@ void Stepper::set_directions() {
  *  2000     1 KHz - sleep rate
  *  4000   500  Hz - init rate
  */
-ISR(TIMER1_COMPA_vect) { Stepper::isr(); }
+ISR(TIMER1_COMPA_vect) {
+  #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
+    Stepper::ISRhandler();
```

Comment:
I think `Stepper::ISRhandler` is short enough that you can just put its code here and drop the extra method call.